rest-backend/src/main/resources/db/migration/V20200219.1130__dropDepIndexes.sql

```diff
@@ -0,0 +1,4 @@
+DROP INDEX if exists public.dep_app_lib_index;
+DROP INDEX if exists public.dep_app_lib_parent_index;
+DROP INDEX if exists public.dep_app_lib_relpath_index;
+alter table app_dependency add constraint UKnueog86fts45j2wcql6idbqwn unique (lib, app, parent, relative_path);
```

Comment:
@serenaponta : The FlyWay migration script contains the explicit creation of the unique constraint ... isn't that created by JPA according to the annotation in class `Dependency`?

Reply:
The constraint is created by JPA in case the application is started with the configuration  (in application.properties)

spring.jpa.hibernate.ddl-auto=create

As we have an already running database, the configuration is set to "validate" and thus the constraint has to be created via a migration.

Users setting it up for the first time should be able to decide if they want to use the entire set of migrations or to remove them and set the configuration to "create" on the first run (and "validate" later on). The constraint in the migration comes from the set of SQL statements created by JPA, however note that i didn't test the behavior of entire service with a newly created schema.